runtime/near-vm-runner/src/wasmer1_runner.rs

```diff
@@ -0,0 +1,252 @@
+            CompilationError::WasmerCompileError { msg: self.to_string() },
+        ))
+    }
+}
+
+impl IntoVMError for wasmer::InstantiationError {
+    fn into_vm_error(self) -> VMError {
+        match self {
+            wasmer::InstantiationError::Link(e) => {
+                VMError::FunctionCallError(FunctionCallError::LinkError { msg: e.to_string() })
+            }
+            wasmer::InstantiationError::Start(e) => e.into_vm_error(),
+            wasmer::InstantiationError::HostEnvInitialization(_) => {
+                VMError::FunctionCallError(FunctionCallError::CompilationError(
+                    CompilationError::PrepareError(PrepareError::Instantiate),
+                ))
+            }
+        }
+    }
+}
+
+impl IntoVMError for wasmer::RuntimeError {
+    fn into_vm_error(self) -> VMError {
+        // These vars are not used in every cases, however, downcast below use Arc::try_unwrap
+        // so we cannot clone self
+        let error_msg = self.message();
+        let trap_code = self.clone().to_trap();
+        match self.downcast::<VMLogicError>() {
+            Ok(e) => e.into_vm_error(),
+            _ => {
+                if let Some(trap_code) = trap_code {
+                    // A trap
+                    VMError::FunctionCallError(FunctionCallError::Wasmer1Trap(
+                        trap_code.to_string(),
+                    ))
+                } else {
+                    // A general error
+                    VMError::FunctionCallError(FunctionCallError::WasmerRuntimeError(error_msg))
+                }
+            }
+        }
+    }
+}
+
+impl IntoVMError for wasmer::ExportError {
+    fn into_vm_error(self) -> VMError {
+        match self {
+            wasmer::ExportError::IncompatibleType => VMError::FunctionCallError(
+                FunctionCallError::MethodResolveError(MethodResolveError::MethodInvalidSignature),
+            ),
+            wasmer::ExportError::Missing(_) => VMError::FunctionCallError(
+                FunctionCallError::MethodResolveError(MethodResolveError::MethodNotFound),
+            ),
+        }
+    }
+}
+
+fn check_method(module: &Module, method_name: &str) -> Result<(), VMError> {
+    let info = module.info();
+    use wasmer_types::ExportIndex::Function;
+    if let Some(Function(index)) = info.exports.get(method_name) {
+        let func = info.functions.get(index.clone()).unwrap();
+        let sig = info.signatures.get(func.clone()).unwrap();
+        if sig.params().is_empty() && sig.results().is_empty() {
+            Ok(())
+        } else {
+            Err(VMError::FunctionCallError(FunctionCallError::MethodResolveError(
+                MethodResolveError::MethodInvalidSignature,
+            )))
+        }
+    } else {
+        Err(VMError::FunctionCallError(FunctionCallError::MethodResolveError(
+            MethodResolveError::MethodNotFound,
+        )))
+    }
+}
+
+pub fn run_wasmer1<'a>(
```

Comment:
Significant amount of copy-paste, but not sure what could be done here.

Reply:
Yeah run_wasmer1 is structurally very alike to run_wasmer, but every step a different type of wasmer instance, module, etc. is used between 0.x and 1.x, so it's difficult to combine them